packages/repository/src/model.ts

```diff
@@ -138,6 +139,7 @@ export class ModelDefinition {
 export abstract class Model {
   static modelName: string;
   static definition: ModelDefinition;
+  static relations: RelationMap;
```

Comment:
Please note that `ModelDefinition` already has `relations` property, see https://github.com/strongloop/loopback-datasource-juggler/blob/fa0039c6a8795d61c162c48160f03396b566cf44/types/model.d.ts#L64

```
export declare class ModelDefinition extends EventEmitter implements Schema {
  name: string;
  properties: AnyObject;
  rawProperties: AnyObject;
  settings?: AnyObject;
  relations?: AnyObject[];
  // ...
}
```

Have you considered reusing it here?

My main concern is about having multiple sources of truth about relations, because it's easy for these places to get out of sync.

Reply:
My thought with `ModelDefinition`'s relations property is that it is used by the old juggler. Since we're coming up with lb4's own relations engine, I thought it'd make sense to separate the implementation of new relations from the old.

It is concerning that the UX here may be confusing for the users though

Reply:
My views are similar to @shimks, what do you think @raymondfeng?

Reply:
Here are few important aspects to consider: 

 - Relation metadata stored in juggler's `ModelDefinition` is used by juggler to (optionally) add foreign key property to the target model, e.g. if Category has many Product models, then we need to define `Product.categoryId` property. Loosely related: #1475

 - Autoupdate/automigrate functionality should create foreign key constraint in the database (I am not sure if our connectors do that though). To do so, they need relations to be defined in juggler's `ModelDefinition` object.

I am ok to leave this discussion out of scope of this pull request if you prefer, as long as we resolve it by DP3.

Reply:
@bajtos I think I'm ok to move the relations metadata back to the one in `ModelDefinition` provided that the data is only used to optionally add foreign key property to the target model. With #1475, I'm guessing this would not matter with lb4 since we'd require models to be 'strict'.
@raymondfeng Seems like we have to do it this way considering the autoupdate/automigrate feature

Reply:
Thanks for raising those points @bajtos. In that light, I am also fine to have the relation metadata stored in the model definition. If we do intend to support autoupdate/automigrate from LB3, then this is the way to go.